orca-clouddriver-provider-titus/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/job/TitusRunJobStageDecorator.java

```diff
@@ -40,4 +60,25 @@ public void modifyDestroyJobContext(
     destroyContext.put("jobId", context.getJobStatus().getId());
     destroyContext.remove("jobName");
   }
+
+  @Alpha
+  private void remoteStageExecution(StageExecution stageExecution) {
+    boolean isRemoteStage = stageExecution
+        .getContext()
+        .getOrDefault("remoteStage", "false")
+        .toString()
+        .equalsIgnoreCase("true");
+
+    if (isRemoteStage) {
+      RemoteStageExecution remoteStageExecution = remoteStageExecutionProvider.get(stageExecution);
+
+      try {
+        TitusPreconfiguredJobProperties.Cluster titusCluster = objectMapper.convertValue(stageExecution.getContext().get("cluster"), TitusPreconfiguredJobProperties.Cluster.class);
+        titusCluster.getEnv().put("REMOTE_STAGE_EXECUTION", Base64.getEncoder().encodeToString(objectMapper.writeValueAsBytes(remoteStageExecution)));
```

Comment:
To the above point ... any know limitations on this? Will this also result in the current stage context getting blown up?

Reply:
Yeah, 16k is the total space available for declared env variables to be injected (32kb is the total overall size for env variables).

I think there's probably some value in limiting what we pass to the remote stage, and then providing some mechanism for querying for upstream stage output.

Reply:
Definitely some value in limiting (may also vary across runtimes).